app/vmalert/notifier/alert.go

```diff
@@ -199,5 +200,10 @@ func (a Alert) toPromLabels(relabelCfg *promrelabel.ParsedConfigs) []prompbmarsh
 		labels = relabelCfg.Apply(labels, 0)
 	}
 	promrelabel.SortLabels(labels)
-	return labels
+
```

Comment:
`promrelabel.SortLabels(labels)` doesn't make sense anymore, because data goes back into map.
Ideally, labels shouldn't be transformed at all. Currently, we do `map => slice => map`, which does not sound great. Have you thought of ways to avoid this?

Reply:
Sorted labels are not mandatory when sending alerts to Alertmanager, since AM will sort them while [calculating the fingerprint of each alert](https://github.com/prometheus/common/blob/b1880d0dabb633dbf29b999c6a046637efb602fe/model/signature.go#L54). But it's better to sort labels on vmalert side as before. 
And you're right, the new map has no benefit here. 
I reverted to the old logic, to drop alerts without labels when forming the write request. It can't trace alerts dropped by relabeling, but we can add that later if needed.